lib/components/component.dart

```diff
@@ -72,6 +78,9 @@ abstract class Component {
   /// Called right before the component is removed from the game
   void onRemove() {}
 
+  /// Called to check whether the point should be counted as a tap on the component
+  bool checkOverlap(Vector2 point) => false;
```

Comment:
This is a result of an API that is spread across several places. We have check overlap here, which means anything since we can't assume that every component subclass instance has a size or even is painted on the screen at all. The doc refers to a tap but there is nothing regarding taps in this class. This is fashionable for the tappable API but with no sense for all the other use cases.

Reply:
This one I agree that it should be moved, but question is what a layer in between `Component` and `PositionComponent` should be called then, `RenderedComponent` maybe?